src/AmalgamatorSession.ts

```diff
@@ -637,12 +637,19 @@ export class AmalgamatorSession extends LoggingDebugSession {
         response: DebugProtocol.Response,
         args: any
     ): Promise<void> {
-        // XXX: Which childDap to send a customRequest to? Probably needs domain specific knowledge.
-        const childResponse = await this.childDaps[0].customRequest(
-            command,
-            args
-        );
-        response.body = childResponse.body;
-        this.sendResponse(response);
+        if (command === 'cdt-gdb-adapter/Memory') {
```

Comment:
The cdt-amalgamator can receive a command called `cdt-amalgamator/Memory` which has the extra info (the child) and pass that child to the `cdt-gdb-adapter/Memory` command.

This would resolve the issue of trying to add a field to cdt-gdb-adapter that is not relevant.

Reply:
You haven't updated it to my suggestion. You are still doing `if (command === 'cdt-gdb-adapter/Memory') {`